### PR TITLE
Exposing external base image properties (as provider fields)

### PR DESCRIPTION
**Pull request by or-shachar**
This is an alternative to #1506 .
Here's an [example to usage](https://github.com/or-shachar/export_base_image_example/tree/Alt).

I wonder if it's okay that we added more attributes to `container_import` (is this a rule that is only called by generated BUILD files of `container_pull`?).

### CODE REVIEW

**Comment by or-shachar** [comment]
Another alternative: Maybe instead of adding fields to `ImportInfo` we should introduce yet another provider called `PullInfo` and populate it only if all 3 fields are provided.

**Comment by or-shachar** [comment]
Sure thing. Agreed. 

Re-custom test rule - You mean actual custom bazel rule? Do you have an example?

**Comment by or-shachar** [comment]
Ok @smukherj1  - I think I fixed all according to your kind comments.

> - [x] I think I like your suggestion of returning a PullInfo provider that container_import returns only if called by container_pull, i.e., all the base image attributes are populated.
done

> - [x] I would call the new attributes base_image_* instead of source_* because I think base_image_* is a term more commonly used to denote the base of a Docker image.
done

> - [x] For tests, I expect you'll need to create a custom test rule that takes a container_pull as a dependency, ensures it produces the new provider and assert the values in the given provider.
Hopefully I got it right... created a custom tests that relies on a minimal `container_pull` that I had to add.

I see that buildkite / RBE is failing for some digest not around my changes.

LMK what you think :-)

File: WORKSPACE [hunk 1]
```diff
@@ -531,3 +531,7 @@ load("@io_bazel_rules_kotlin//kotlin:kotlin.bzl", "kotlin_repositories", "kt_reg
 kotlin_repositories()
 
 kt_register_toolchains()
+
+load("//tests/container:pull_info_validation_test.bzl", "define_base_image_for_tests")
+
+define_base_image_for_tests()
```
[the remark this answers is not in the document]
Comment (or-shachar) [reply WORKSPACE]: ok ... changed it accordingly but now I cannot load the base image properties from one place. eg: if someone changes the digest of `distroless_base_fixed_id` in `WORKSPACE` file  - they would also need to change it in the BUILD file.

**Comment by or-shachar** [comment]
@smukherj1  - thanks for your review. Fixed.

**Comment by smukherj1** [comment]
Ok this PR LGTM. You'll need to wait for #1513 to fix the sha256 issue and #1495 which was requested before this PR. So my plan is as follows:
1. Prioritize merging 1513 as it fixes the PR failure
2. For for 1495 owner to rebase until tomorrow. If so I'll merge that first.
3. Merge this PR.

**Comment by or-shachar** [comment]
Thanks! btw - Really appreciate the transparency ! 🌹

**Comment by smukherj1** [comment]
@or-shachar this PR and #1509 can now be merged. But they'll need to be re-based/merged with HEAD first. Please do that for whichever PR you want to have merged first.

**Comment by or-shachar** [comment]
I've rebased #1509 for now.
BTW - about this PR - do you prefer me to squash all commits or you perform this yourself?

**Comment by smukherj1** [comment]
> I've rebased #1509 for now.
> BTW - about this PR - do you prefer me to squash all commits or you perform this yourself?

Our repo's merge commit policy is set to squash commits I believe. So it doesn't really matter whether you squash or not. I would just rebase or merge HEAD into your branch (once #1509) is merged.

**Comment by or-shachar** [comment]
Sorry @smukherj1  - I've just rebased

**Pull request merged**